### ISSUE TITLE
Allowed choice of variant product in product picker window after search with not allowed variants

### ISSUE THREAD

**Issue by thirdknown** [shopsys/shopsys#1187]
### What is happening
I use `Shopsys\FrameworkBundle\Form\ProductsType` in form with option `allow_variants` set to `false`.

So when I open products picker window, I can't choose variant - it is correct behavior.

But when I use search in that same product picker window, I can choose variant from searched results - it is not correct behavior, because I set `allow_variants` to `false`.

### Expected result
I can't choose variant product after use search in product picker window (with using `ProductsType` with option `allow_variants` set to `false`).

**Comment by vitek-rostislav** [comment]
Hi @thirdknown, thanks for reporting :ok_hand: 

I did some digging - the value for `allow_variants` parameter is obtained from the request when the popup is opened, see `ProductPickerController::pickMultipleAction` and search for `data-product-picker-url` in `framework/src/Resources/views/Admin/Form/theme.html.twig`. 

However, when the search form is submitted, the data are lost from the request and hence the bug occurs :disappointed: 

When the popup is opened, the request URI looks like this:
```
/admin/product-picker/pick-multiple/0/?allowMainVariants=1&allowVariants=0
```

When the search is submitted, the request URI looks like this:
```
/admin/product-picker/pick-multiple/0/?quick_search_form%5Btext%5D=Apple
```

The same applies to the multiple, and single selection of products.

**Comment by pk16011990** [comment]
And resetting of the form removes GET parametrs too. It is not solved in #1311 :-/

However solution can be adding `allowMainVariants` and `allowVariants` into route parameters and It will not be as GET parameters.

**Comment by pk16011990** [comment]
remove stale

**Comment by s3tezsky** [comment]
Fixed in #1966
